### DISCUSSION TITLE
Add option to display highest combo in combo counter

### DISCUSSION THREAD

**Discussion by nanashi-1** [ppy/osu#20631] [Ideas] [answered]
The combo counter right now displays the current combo. Is adding an option in the skin editor to make the combo counter display the highest combo a good idea?

**Comment by mk56-spn** [comment]
easily doable, i wonder if this should be a subsetting or its own component

**Reply by mk56-spn** [reply]
https://user-images.githubusercontent.com/74463310/194712704-cf8be578-1c5c-43a5-8613-6e13029f58d9.mov

**Reply by apollo-dw** [reply]
this would be ideal as a subsetting IMO

**Reply by mk56-spn** [reply]
https://user-images.githubusercontent.com/74463310/194720320-8dff1024-6213-4a60-b06e-601216ce879f.mov


does this need a test?

**Reply by ghost** [reply]
i think a checkbox titled "display highest combo" would work better

**Reply by nanashi-1** [reply]
> 2022-10-08.19-33-30.mov
> 
> does this need a test?

yes

**Reply by nanashi-1** [reply]
we may need a member to approve this

**Reply by mk56-spn** [reply]
> we may need a member to approve this

Yep. Not always seemingly required but I think I'll just wait until one of them comments in whether this is wanted before I PR to avoid pointless PRs

**Reply by peppy** [reply]
This should not be a setting on the existing component. It causes confusion as there's no way to discern it from the standard combo. It will need a design which clearly distinguishes it as "max".

**Reply by mk56-spn** [reply]
![max](https://user-images.githubusercontent.com/74463310/194907841-3e672ced-4c73-473d-94e9-790b7140195b.png)

im kinda liking the idea of showing it like this ( i already redid the hierarchy, two seperate classes that share logic through a parent combocounter.cs) not sure about going with "max" nomenclature, feels like that would be the max combo available on the map.  "max achieved"?

**Reply by peppy** [reply]
maybe "best combo"? on "longest combo"?

**Reply by mk56-spn** [reply]
longest combo is perfect, went with that, last couple of things:

are you ok with it being a slightly different colour as shown above and here, for the sake of differentiating them a tad more?

should i change the combo counter text to be "fixedwidth"? its used in pretty much all other counters and imo looks a lot better than this : 


https://user-images.githubusercontent.com/74463310/194925913-a88ec370-ec1d-47c8-9b09-cd55655a4966.mov

**Reply by peppy** [reply]
I think this is fine for now. I may tweak the colour after you PR it.

Fixed width may be okay, will need to check kerning and alignment in the HUD.

**Comment by mk56-spn** [comment, marked as answer]
https://github.com/ppy/osu/pull/20724 closes this
